src/frontend/citizen-portal/src/app/components/dispute-stepper/dispute-stepper.component.ts

```diff
@@ -314,6 +352,30 @@ export class DisputeStepperComponent implements OnInit, AfterViewInit {
     }
   }
 
+  /**
+   * Handles the change of the signature type.
+   *
+   * @param {DisputeSignatoryType} value - The new value of the signature type.
+   */
+  onChangeSignatureType(value: DisputeSignatoryType) {
+    if (value === this.SignatoryType.D) {
+      this.signatureBoxForm.controls.disputant_signatory_name.setValidators([Validators.maxLength(100), Validators.required]);
+      this.signatureBoxForm.controls.agent_signatory_name.setValue(null);
+      this.signatureBoxForm.controls.agent_signatory_name.clearValidators();
+      this.signatureBoxForm.controls.agent_signatory_name.disable();
+      this.signatureBoxForm.controls.disputant_signatory_name.enable();
+    } else if (value === this.SignatoryType.A) {
+      this.signatureBoxForm.controls.agent_signatory_name.setValidators([Validators.maxLength(100), Validators.required]);
+      this.signatureBoxForm.controls.disputant_signatory_name.setValue(null);
+      this.signatureBoxForm.controls.disputant_signatory_name.clearValidators();
+      this.signatureBoxForm.controls.disputant_signatory_name.disable();
+      this.signatureBoxForm.controls.agent_signatory_name.enable();
+    }
```

Comment:
do we need an else {} block here to reset both agent and disputant values?

Reply:
I don't think we need that here because user has to select one or the other of signatory box is visible. There is another check that resets both agent and disputant values on line 296 if the RequestCourtAppearance value is "Y". (If user selects the option to request court appearance instead of written reasons)